<launch>
  <arg name="use_rp" default="true"/>
  <arg name="use_velo" default="true"/>
  <arg name="use_joy" default="true"/>

  <include file="$(find leo_navigation)/launch/gmapping.launch"/>
  <include if="$(arg use_joy)" file="$(find teleop_twist_joy)/launch/teleop.launch"/>
  <include if="$(arg use_velo)" file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/>
  <include file="$(find leo_navigation)/launch/odometry.launch"/>
  
  <node if="$(arg use_rp)" name="rplidarNode_front"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
    <param name="serial_port"         type="string" value="/dev/rplidar"/>
    <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
    <!--param name="serial_baudrate"     type="int"    value="256000"--><!--A3 -->
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    <param name="output_scan"         type="string" value="/up_scan"/>
  </node>

  <node pkg="leo_driving" type="leo_driving_node" name="leo_driving_node" output="screen">
	<param name="Kpy_param" value="2.5"  /> 
	<param name="linear_vel" value="0.2" /> 
	<param name="boundary_percent" value="0.02" /> 
	<param name="side_obstacle_dist" value="0.2" /> 
	<param name="front_obstacle_dist" value="0.5" /> 
	<param name="obastalce_coefficient" value="0.1" /> 

	<param name="gmapping_mode" value="1.0" /> 	
	<param name="amcl_mode" value="0.0" /> 
  </node>
</launch>

